ciao-cli/template.go

```diff
@@ -61,6 +70,40 @@ func findField(fieldPath []string, v reflect.Value) reflect.Value {
 	return f
 }
 
+func filterBySubstr(obj interface{}, field, val string) (retval interface{}) {
```

Comment:
This function is almost identical to filterByField.  Perhaps we need to make a new function, fieldField that takes a function as a parameter, e.g.,

```
func filterField(obj interface{}, field, val string, cmp func(string,string) bool)(retval interface{}) {
```

filterBySubstr then becomes

```
func filterBySubstr(obj interface{}, field, val string) (retval interface{}) {
     return filterField(obj, field, val, strings.Contains)
}
```

filterByField

```
func filterByField(obj interface{}, field, val string) (retval interface{}) {
     return filterField(obj, field, val, func(a, b string) bool {
         return a == b
    })
}

```

You could then have a whole file of filter functions, with now extra effort, e.g,

filter
filterSubstr (might be better to call is filterContains to match the std lib function called)
filterHasPrefix
filterHasSuffix
filterRegexp
filterFolded